zboxcore/zboxutil/http.go

```diff
@@ -562,18 +559,42 @@ func MakeSCRestAPICall(scAddress string, relativePath string, params map[string]
 			entityResult[sharder] = retObj
 		}
 
-		var rate = maxCount * 100 / float32(numSharders)
+		var rate = float32(maxCount*100) / float32(numSharders)
 		if rate >= consensusThresh {
 			break // got it
 		}
 	}
 
 	var err error
-	rate := maxCount * 100 / float32(numSharders)
+	rate := float32(maxCount*100) / float32(numSharders)
 	if rate < consensusThresh {
 		err = errors.New("consensus_failed", "consensus failed on sharders")
 	}
 
+	c := 0
+	dominant := 200
+	for code, count := range responses {
```

Comment:
@dabasov I am not following logic here, does it means if any of sharders giving response code != 200 it will fail?

Reply:
@moldis look, this code will chose the most popular return code, and will use it, it's like weak consensus, Imagine you have 6 sharders and 3 returned 400, 2 returned 200 and 1 returned 500, then 400 will be chosen as dominant.